src/converters/BindingConverters.ts

```diff
@@ -22,16 +22,70 @@ export function getBindingDefinitions(info: FunctionInfo): BindingDefinition[] {
 }
 
 export function getNormalizedBindingData(request: rpc.IInvocationRequest): Dict<any> {
+
+  // Repopulate triggerMetaData body and rawBody if they're not already populated.
+  var [inputDataBody, inputDataRawBody] = getBodyAndRawBody(request.inputData);
+  populateTriggerMetadataIfNotPopulated(request.triggerMetadata, inputDataBody, inputDataRawBody);
```

Comment:
Is there a reason you're only removing the body/rawBody instead of the whole http request? the whole thing will be duplicated.

Reply:
That was my initial thought too. But on second thoughts I thought it might be better/safer to remove (and subsequently reconstruct on the worker side)  the bare minimum while trying to achieve huge perf gains. Besides, excluding body and rawbody, the rest of the elements are magnitudes of times smaller anyway.

In other words, I thought this was a lot less riskier than removing everything and trying to reconstruct it all on the worker side.

Please let me know if you have different thoughts around this and/or if I'm missing something

Reply:
Host should just stop sending DefaultHttpRequest as part of trigger metadata. Lets put this behind capabilities flag. That will allow you to merge this change without host changes.

Reply:
synced up offline about this. Changed this to the initial thought process of just reconstructing everything instead of just the body and the rawBody. Added this under capability flag and added unit test too.